modyn/selector/internal/selector_strategies/downsampling_strategies/downsampling_scheduler.py

```diff
@@ -78,26 +79,17 @@ def inform_next_trigger(self, next_trigger_id: int, selector_storage_backend: Ab
 
 
 def instantiate_scheduler(
-    config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int
+    config: CoresetSelectionConfig, modyn_config: dict, pipeline_id: int
 ) -> DownsamplingScheduler:
-    if config.get("downsampling_config") is None:
-        # missing downsampler, use NoDownsamplingStrategy
-        list_of_downsamplers = [{"strategy": "No"}]
-        list_of_thresholds: list[int] = []
-    elif config["downsampling_config"].get("downsampling_list") is None:
+    if isinstance(config.downsampling_config, MultiDownsamplingConfig):
```

Comment:
I think there is some confusing overload here. `MultiDownsamplingConfig` lives (currently) on a different abstraction level than the actual downsampling configs if I understand correctly. It lives on the pipeline config level. Now here we use it to check whether we have a mutlidownsamplingconfig or a single one. I think (even if it is just naming) something is a bit inconsistent here. I think downsampling_config is either the downsampler itself or meta information for downsampling, but right now it can be somewhat both? I am a little confused actually.

Reply:
I just moved `MultiDownsamplingConfig` also to the downsampling config file.

I understand this confusion. This exists before this PR (in main branch) https://github.com/eth-easl/modyn/blob/6c3d63f22e98f41d3bcbd0c0cc2fcd1053abdc4f/modyn/config/schema/pipeline.py#L228.

I can rename the pydantic model class `DownsamplingConfig` to `SingleDownsamplingConfig`. How do you like it?

Reply:
I am not fully sure about the semantics behind `MultiDownsamplingConfig`, but why can't we use
` downsampling_config: list[DownsamplingConfig] = Field(default_factory=[] , ...)`

Reply:
Because `MultiDownsamplingConfig` is more than a list of downsamplers :) See the definition in the pipeline.py